app/controllers/chores_controller.rb

```diff
@@ -0,0 +1,56 @@
+class ChoresController < ApplicationController
+  def create
+    title = params[:title]
+    description = params[:description]
+    due_date = params[:due_date]
+    assignee_email = params[:assignee_email]
+    hideout_id = params[:hideout_id]
+
+    if title.nil? || hideout_id.nil?
+      render status: 400
+    else
+      if assignee_email.nil?
+        assignee_id = nil
+      else
+        assignee_id = User.find_by(email: assignee_email).id
+      end
+      chore = Chore.create(title: title, description: description, due_date: due_date, assignee_id: assignee_id, hideout_id: hideout_id)
+      chore.save
+      render status: :created, :json => {title: title, description: description, due_date: due_date, assignee_email: assignee_email, hideout_id: hideout_id}
+    end
+  end
+
+  def update
+    params.require(:id)
+    id = params[:id]
+
+    # TODO: add middleware for checking if user exists and chores exists 
+
+    chore = Chore.find_by(id: id)
+
+    if params.has_key?(:title)
+      chore.update(title: params[:title])
+    end
+    if params.has_key?(:description)
+      chore.update(description: params[:description])
+    end
+    if params.has_key?(:due_date)
+      chore.update(due_date: params[:due_date])
+    end
+    if params.has_key?(:assignee_email)
+      chore.update(assignee_id: User.find_by(email: params[:assignee_email]).id)
+    end
```

Comment:
Instead of having a series of if-statements, it would be better to just take in the entire JSON Chore object that the user provides and just replace the chore object with that. We wouldn't really make a request from the client side with only one attribute changed, rather just the entire modified object (regardless of whether or not certain attributes are changed).